examples/pytorch/audio-classification/run_audio_classification.py

```diff
@@ -0,0 +1,379 @@
+    )
+
+    log_level = training_args.get_process_log_level()
+    logger.setLevel(log_level)
+    transformers.utils.logging.set_verbosity(log_level)
+    transformers.utils.logging.enable_default_handler()
+    transformers.utils.logging.enable_explicit_format()
+
+    # Log on each process the small summary:
+    logger.warning(
+        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu} "
+        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
+    )
+    logger.info(f"Training/evaluation parameters {training_args}")
+
+    # Set seed before initializing model.
+    set_seed(training_args.seed)
+
+    # Detecting last checkpoint.
+    last_checkpoint = None
+    if os.path.isdir(training_args.output_dir) and training_args.do_train and not training_args.overwrite_output_dir:
+        last_checkpoint = get_last_checkpoint(training_args.output_dir)
+        if last_checkpoint is None and len(os.listdir(training_args.output_dir)) > 0:
+            raise ValueError(
+                f"Output directory ({training_args.output_dir}) already exists and is not empty. "
+                "Use --overwrite_output_dir to train from scratch."
+            )
+        elif last_checkpoint is not None and training_args.resume_from_checkpoint is None:
+            logger.info(
+                f"Checkpoint detected, resuming training at {last_checkpoint}. To avoid this behavior, change "
+                "the `--output_dir` or add `--overwrite_output_dir` to train from scratch."
+            )
+
+    # Initialize our dataset and prepare it for the audio classification task.
+    raw_datasets = DatasetDict()
+    raw_datasets["train"] = load_dataset(
+        data_args.dataset_name, data_args.dataset_config_name, split=data_args.train_split_name
+    )
+    raw_datasets["eval"] = load_dataset(
+        data_args.dataset_name, data_args.dataset_config_name, split=data_args.eval_split_name
+    )
+
+    if data_args.audio_column_name not in raw_datasets["train"].column_names:
+        raise ValueError(
+            f"--audio_column_name {data_args.audio_column_name} not found in dataset '{data_args.dataset_name}'. "
+            "Make sure to set `--audio_column_name` to the correct audio column - one of "
+            f"{', '.join(raw_datasets['train'].column_names)}."
+        )
+
+    if data_args.label_column_name not in raw_datasets["train"].column_names:
+        raise ValueError(
+            f"--label_column_name {data_args.label_column_name} not found in dataset '{data_args.dataset_name}'. "
+            "Make sure to set `--label_column_name` to the correct text column - one of "
+            f"{', '.join(raw_datasets['train'].column_names)}."
+        )
+
+    feature_extractor = AutoFeatureExtractor.from_pretrained(
+        model_args.feature_extractor_name or model_args.model_name_or_path,
+        return_attention_mask=True,
```

Comment:
We should change this here. If we change the fetaure extractor config by passing `return_attention_mask=True` then we also have to save the feature extractor with the config having `return_attention_mask=True`. Otherwise people will train: https://huggingface.co/facebook/wav2vec2-base/blob/main/preprocessor_config.json (which has `return_attention_mask==False`), not save it and use the model with the default feature extractor which will lead to incorrect results. 

The problem with wav2vec2-base is that batched inference != non-batched inference by model design. In my experiments of wav2vec2-base on ASR, I found that not using the attention mask led to better results. 
Think we have to options here:

1) Not pass `return_attention_mask=True` - we would probably have to quickly re-run the experiments to see if the results are good with this as well

2) Save the feature extractor alongside the model

Reply:
In this case the feature extractor is passed as `tokenizer=feature_extractor` to `Trainer`, so it's saved alongside the model: https://huggingface.co/anton-l/wav2vec2-base-langid/blob/main/preprocessor_config.json

I've added an `--attention_mask` flag and re-done the CommonLanguage experiment with `--attention_mask=False`, and the result is indeed slightly better :)

Reply:
Sorry, you're 100% right - the `Trainer` actually saves the `feature_extractor`! Ok in this case - all good for me :-)